cloudinit/cmd/status.py

```diff
@@ -196,6 +196,54 @@ def get_bootstatus(disable_file, paths) -> Tuple[UXAppBootStatusCode, str]:
     return (bootstatus_code, reason)
 
 
+def _get_systemd_status() -> Optional[UXAppStatus]:
+    """Get status from systemd.
+
+    Using systemd, we can get more fine-grained status of the
+    individual unit. Determine if we're still
+    running or if there's an error we haven't otherwise detected
+    """
+    for service in [
+        "cloud-final.service",
+        "cloud-config.service",
+        "cloud-init.service",
+        "cloud-init-local.service",
+    ]:
+        stdout = subp.subp(
+            [
+                "systemctl",
+                "show",
+                "--property=ActiveState,UnitFileState,SubState",
+                service,
+            ],
+        ).stdout
+        states = dict(
+            [[x.strip() for x in r.split("=")] for r in stdout.splitlines()]
+        )
+        if not (
+            states["UnitFileState"].startswith("enabled")
+            or states["UnitFileState"] == "static"
+        ):
+            # Service is not set to run, so don't determine state
+            # based on this service
+            continue
+        if (
+            states["ActiveState"] == "active"
+            and states["SubState"] == "exited"
+        ):
+            # Service exited normally, nothing interesting from systemd
+            continue
```

Comment:
If `cloud-init-final.service` exited normally, do we really want to loop through all prior required boot stages? Or can we say, last stage of cloud-init has exited success, so we are done.

Reply:
I realize it's unlikely, but I was thinking it might be possible for an earlier service file to exit with error in a way that might not make it's way to status or result jsons. Not something we should worry about?

Reply:
works for me. I think it's possible that we can have a failed earlier stage and still execute final etc. So I may be incorrect in that assumptions. Ship it